apps/admin/backend/src/app.test.ts

```diff
@@ -496,3 +497,170 @@ test('auth before election definition has been configured', async () => {
+ * @param errorStringToFailWith - a human-readable string that store will fail with. This helps identify which test is failing.
+ * @param setUpFailure - a function that accepts a `Workspace` and replaces a function on that Workspace with a stub that throws.
+ */
+async function runStoreFailureTest(
+  errorStringToFailWith: string,
+  setUpFailure: (workspace: Workspace) => void
+) {
+  const { apiClient, auth, workspace, logger } = buildTestEnvironment();
+  const { electionDefinition, systemSettings } =
+    electionMinimalExhaustiveSampleFixtures;
+  await configureMachine(apiClient, auth, electionDefinition);
+  mockSystemAdministratorAuth(auth);
+
+  setUpFailure(workspace);
+
+  const result = await apiClient.setSystemSettings({
+    systemSettings: systemSettings.asText(),
+  });
+  expect(result.isOk()).toEqual(false);
+  expect(result.isErr()).toEqual(true);
+
+  const err = result.err();
+  assert(err);
+  expect(err.type).toEqual('database');
+  expect(err.message).toEqual(
+    `Failed to save system settings to db. Error: ${errorStringToFailWith}`
+  );
+
+  // Logger call 1 is made by configureMachine when loading the election definition
+  expect(logger.log).toHaveBeenNthCalledWith(
+    2,
+    LogEventId.SystemSettingsSaveInitiated,
+    'system_administrator',
+    { disposition: 'na' }
+  );
+  expect(logger.log).toHaveBeenNthCalledWith(
+    3,
+    LogEventId.SystemSettingsSaved,
+    'system_administrator',
+    { disposition: 'failure', error: errorStringToFailWith }
+  );
+}
+
+test('setSystemSettings returns an error when store.addSystemSettings fails', async () => {
+  const exampleErrorString = 'db error at addSystemSettings';
+  await runStoreFailureTest(exampleErrorString, (workspace: Workspace) => {
+    /* eslint-disable no-param-reassign */
+    workspace.store.addSystemSettings = jest.fn(() => {
+      throw new Error(exampleErrorString);
+    });
+  });
+});
+
+test('setSystemSettings returns an error when store.setCurrentSystemSettingsId fails', async () => {
+  const exampleErrorString = 'db error at setCurrentSystemSettingsId';
+  await runStoreFailureTest(exampleErrorString, (workspace: Workspace) => {
+    /* eslint-disable no-param-reassign */
+    workspace.store.setCurrentSystemSettingsId = jest.fn(() => {
+      throw new Error(exampleErrorString);
+    });
+  });
+});
+
+test('setSystemSettings returns an error for malformed input', async () => {
+  const { apiClient, auth } = buildTestEnvironment();
+
+  const { electionDefinition } = electionMinimalExhaustiveSampleFixtures;
+  await configureMachine(apiClient, auth, electionDefinition);
+
+  mockSystemAdministratorAuth(auth);
+
+  const malformedInput = {
+    invalidField: 'hello',
+  } as const;
+
+  const result = await apiClient.setSystemSettings({
+    systemSettings: JSON.stringify(malformedInput),
+  });
+  expect(result.isOk()).toEqual(false);
+  expect(result.isErr()).toEqual(true);
+
+  const err = result.err();
+  assert(err);
```

Comment:
This could be like this:

```suggestion
  assert(result.isErr());
  const err = result.err();
```

The `assert` will help TS narrow the type of `result` from `Result` to `Err`.

Reply:
https://github.com/votingworks/vxsuite/pull/3147/commits/6a7c3e8b412178f24a94699d9c141ea1ce4e7ac0